src/arch/xtensa/drivers/timer.c

```diff
@@ -103,7 +103,7 @@ uint64_t arch_timer_get_system(struct timer *timer)
 	return time;
 }
 
-int arch_timer_set(struct timer *timer, uint64_t ticks)
+int64_t arch_timer_set(struct timer *timer, uint64_t ticks)
```

Comment:
shouldn't this be uint64_t if we are returning ticks ?

Reply:
@lgirdwood, to be consistent with other .set_timer() functions it needs to be signed integer as some of them returns negative error codes. As for the range, int64_t max value is big enough to hold any reasonable value...unless you want to simulate how SoF will work 731+ years from boot we should be safe :)